src/mongocrypt-opts.c

```diff
@@ -132,6 +134,12 @@ _mongocrypt_opts_kms_providers_validate (
       }
    }
 
+   if (kms_providers->need_credentials &&
+       !opts->use_need_kms_credentials_state) {
+      CLIENT_ERR ("on-demand credentials not supported");
```

Comment:
Rather than "not supported", would "not enabled" be more accurate, given users have the option to opt into using on-demand credentials?

Reply:
Yes. Done.